resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -16,256 +16,130 @@
  */
 package org.apache.spark.deploy.kubernetes.submit
 
-import java.io.File
 import java.util.{Collections, UUID}
 
-import io.fabric8.kubernetes.api.model.{ContainerBuilder, EnvVarBuilder, OwnerReferenceBuilder, PodBuilder, QuantityBuilder}
+import io.fabric8.kubernetes.api.model.{ContainerBuilder, HasMetadata, OwnerReferenceBuilder, PodBuilder}
 import io.fabric8.kubernetes.client.KubernetesClient
-import scala.collection.JavaConverters._
+import scala.collection.mutable
 
 import org.apache.spark.SparkConf
-import org.apache.spark.deploy.kubernetes.{ConfigurationUtils, SparkKubernetesClientFactory}
 import org.apache.spark.deploy.kubernetes.config._
 import org.apache.spark.deploy.kubernetes.constants._
-import org.apache.spark.deploy.rest.kubernetes.ResourceStagingServerSslOptionsProviderImpl
+import org.apache.spark.deploy.kubernetes.SparkKubernetesClientFactory
+import org.apache.spark.deploy.kubernetes.submit.submitsteps.{KubernetesDriverSpec, KubernetesSubmissionStep}
 import org.apache.spark.internal.Logging
-import org.apache.spark.internal.config.ConfigEntry
-import org.apache.spark.launcher.SparkLauncher
 import org.apache.spark.util.Utils
 
-/**
- * Submission client for launching Spark applications on Kubernetes clusters.
- *
- * This class is responsible for instantiating Kubernetes resources that allow a Spark driver to
- * run in a pod on the Kubernetes cluster with the Spark configurations specified by spark-submit.
- * The API of this class makes it such that much of the specific behavior can be stubbed for
- * testing; most of the detailed logic must be dependency-injected when constructing an instance
- * of this client. Therefore the submission process is designed to be as modular as possible,
- * where different steps of submission should be factored out into separate classes.
- */
+private[spark] case class ClientArguments(
+     mainAppResource: MainAppResource,
+     otherPyFiles: Seq[String],
+     mainClass: String,
+     driverArgs: Array[String])
+
+private[spark] object ClientArguments {
+  def fromCommandLineArgs(args: Array[String]): ClientArguments = {
+    var mainAppResource: Option[MainAppResource] = None
+    var otherPyFiles = Seq.empty[String]
+    var mainClass: Option[String] = None
+    val driverArgs = mutable.Buffer.empty[String]
+    args.sliding(2).toList.collect {
+      case Array("--primary-py-file", mainPyFile: String) =>
```

Comment:
Yep - the contract for arguments sent to the child submission client class have changed.